tests/src/test/scala/com/nvidia/spark/rapids/BroadcastHashJoinSuite.scala

```diff
@@ -52,17 +55,21 @@ class BroadcastHashJoinSuite extends SparkQueryCompareTestSuite {
 
       for (name <- Seq("BROADCAST", "BROADCASTJOIN", "MAPJOIN")) {
         val plan1 = spark.sql(s"SELECT /*+ $name(t) */ * FROM t JOIN u ON t.longs = u.longs")
-          .queryExecution.executedPlan
         val plan2 = spark.sql(s"SELECT /*+ $name(u) */ * FROM t JOIN u ON t.longs = u.longs")
-          .queryExecution.executedPlan
 
-        val res1 = plan1.find(ShimLoader.getSparkShims.isGpuBroadcastHashJoin(_))
-        val res2 = plan2.find(ShimLoader.getSparkShims.isGpuBroadcastHashJoin(_))
+        val finalplan1 = plan1.queryExecution.executedPlan
+        plan1.collect()
+        val finalPlan1 = findOperator(finalplan1,
+          ShimLoader.getSparkShims.isGpuBroadcastHashJoin(_))
+        assert(ShimLoader.getSparkShims.getBuildSide
+        (finalPlan1.get.asInstanceOf[HashJoin]).toString == "GpuBuildLeft")
 
-        assert(ShimLoader.getSparkShims.getBuildSide(res1.get.asInstanceOf[HashJoin]).toString ==
-          "GpuBuildLeft")
-        assert(ShimLoader.getSparkShims.getBuildSide(res2.get.asInstanceOf[HashJoin]).toString ==
-          "GpuBuildRight")
+        val finalplan2 = plan2.queryExecution.executedPlan
+        plan2.collect()
```

Comment:
Could you add the `// execute the plan so that the final adaptive plan is available when AQE is on` comment here as well

Reply:
done.